src/integ_test_resources/ios/sdk/integration/cdk/cdk_integration_tests_ios/iot_stack.py

```diff
@@ -205,7 +213,78 @@ def setup_custom_authorizer(self):
+            "Manages an asymmetric CMK and token signature for iot custom authorizer with username and password.",
+            token_value)
+
+        custom_authorizer_token_signature = iot_custom_authorizer_key_resource.get_att(
+            "custom_authorizer_token_signature"
+        ).to_string()
+        self._parameters_to_save[
+            "custom_authorizerUserPass_token_signature"
+        ] = custom_authorizer_token_signature
+
+        #Force region to 'us-east-1' due to enhanced custom authorizers only available in this region
+        authorizer_function_arn = self.setup_custom_authorizer_function("2",
+                                                                        "custom_resources/iot_custom_authorizerUserPass_function",
+                                                                        "iot_custom_authorizerUserPass.handler",
+                                                                        "Sample custom authorizer that allows or denies based on username and password",
+                                                                        "us-east-1")
+        create_authorizer_policy = aws_iam.PolicyStatement(
+            effect=aws_iam.Effect.ALLOW, actions=["iot:CreateAuthorizer",
+                                                  "iot:UpdateAuthorizer",
+                                                  "iot:DeleteAuthorizer",
+                                                  "iot:UpdateDomainConfiguration",
+                                                  "iot:CreateDomainConfiguration",
+                                                  "iot:DescribeDomainConfiguration",
+                                                  "iot:DeleteDomainConfiguration"], resources=["*"],
+        )
+        provider_lambda = aws_lambda.SingletonFunction(
+            self,
+            "iot_custom_authorizerUserPass_provider_lambda",
+            uuid="iot_custom_authorizerUserPass_provider_lambda_20200507150213",
+            runtime=aws_lambda.Runtime.PYTHON_3_7,
+            code=aws_lambda.Code.asset("custom_resources/iot_custom_authorizerUserPass_provider"),
+            handler="iot_custom_authorizerUserPass_provider.on_event",
+            description="Sets up an IoT custom authorizer for user password & required domain config due to beta status",
+            current_version_options=aws_lambda.VersionOptions(
+                removal_policy=core.RemovalPolicy.DESTROY
+            ),
+            initial_policy=[create_authorizer_policy],
+        )
+
+        provider = custom_resources.Provider(
+            self, "iot_custom_authorizerUserPass_provider", on_event_handler=provider_lambda
+        )
+
+        public_key = iot_custom_authorizer_key_resource.get_att(
+            "custom_authorizer_public_key"
+        ).to_string()
+
+        iot_endpoint = core.CustomResource(
+            self,
+            "iot_custom_authorizerUserPass",
+            resource_type="Custom::IoTCustomAuthorizer",
```

Comment:
Fancy. I bet this took a while.

Reply:
Hah, not really.. there were other examples in the package I could get inspiration from.